src/App.tsx

```diff
@@ -1,26 +1,336 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  Dispatch,
+  SetStateAction,
+  FormEvent,
+  useMemo,
+} from 'react';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import cN from 'classnames';
+import { TodoItem } from './components/TodoItem';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { ErrorNotification } from './components/ErrorNotification';
+import { filterTodosByStatus } from './utils/filterTodosByStatus';
+import { removeTodoById } from './utils/removeTodoById';
 
-const USER_ID = 0;
+const filterValues = Object.values(Filter);
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Error>(Error.Default);
+  const [activeFilter, setActiveFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoToDeleteIds, setTodoToDeleteIds] = useState<number[]>([]);
+
+  const [inputText, setInputText] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [statusChangeId, setStatusChangeId] = useState<number[]>([]);
```

Comment:
```suggestion
  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
```
One state for todo ids is enough